test/integration/test_kubernetes_runner.py

```diff
@@ -0,0 +1,229 @@
+
+    @skip_without_tool("job_environment_default")
+    def test_job_environment(self):
+        job_env = self._run_and_get_environment_properties()
+        assert job_env.some_env == '42'
+
+    @skip_without_tool('cat_data_and_sleep')
+    def test_kill_process(self):
+        with self.dataset_populator.test_history() as history_id:
+            hda1 = self.dataset_populator.new_dataset(history_id, content="1 2 3")
+            running_inputs = {
+                "input1": {"src": "hda", "id": hda1["id"]},
+                "sleep_time": 240,
+            }
+            running_response = self.dataset_populator.run_tool(
+                "cat_data_and_sleep",
+                running_inputs,
+                history_id,
+                assert_ok=False,
+            ).json()
+            job_dict = running_response["jobs"][0]
+
+            app = self._app
+            sa_session = app.model.context.current
+            external_id = None
+            state = False
+
+            job = sa_session.query(app.model.Job).filter_by(tool_id="cat_data_and_sleep").one()
+            # Not checking the state here allows the change from queued to running to overwrite
+            # the change from queued to deleted_new in the API thread - this is a problem because
+            # the job will still run. See issue https://github.com/galaxyproject/galaxy/issues/4960.
+            max_tries = 60
+            while max_tries > 0 and external_id is None or state != app.model.Job.states.RUNNING:
+                sa_session.refresh(job)
+                assert not job.finished
+                external_id = job.job_runner_external_id
+                state = job.state
+                time.sleep(1)
+                max_tries -= 1
+
+            status = json.loads(subprocess.check_output(['kubectl', 'get', 'job', external_id, '-o', 'json']))
+            assert status['status']['active'] == 1
+
+            delete_response = self.dataset_populator.cancel_job(job_dict["id"])
+            assert delete_response.json() is True
+
+            # Wait for job to be cancelled in kubernetes
+            time.sleep(2)
+            status = json.loads(subprocess.check_output(['kubectl', 'get', 'job', external_id, '-o', 'json']))
+            assert 'active' not in status['status']
+
+    @skip_without_tool('job_properties')
+    def test_exit_code_127(self):
+        inputs = {
+            'failbool': True
+        }
+        running_response = self.dataset_populator.run_tool(
+            "job_properties",
+            inputs,
+            self.history_id,
+            assert_ok=False,
+        )
+        result = self.dataset_populator.wait_for_tool_run(run_response=running_response, history_id=self.history_id, assert_ok=False).json()
+        details = self.dataset_populator.get_job_details(result['jobs'][0]['id'], full=True).json()
+
+        assert details['state'] == 'error', details
+        assert details['stdout'] == 'The bool is not true\n', details
```

Comment:
This test tests the presence of messages on stdout and stderr, so I'd assume this works ?